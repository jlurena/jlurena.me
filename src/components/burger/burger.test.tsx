
import { render, screen, fireEvent } from '@testing-library/react';
import { describe, it, expect, vi } from 'vitest';
import Burger from './burger';

describe('<Burger />', () => {
  it('renders correctly (snapshot)', () => {
    const { asFragment } = render(<Burger onClick={() => {}} onBlur={() => {}} />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('calls onClick handler', () => {
    const onClick = vi.fn();
    render(<Burger onClick={onClick} onBlur={() => {}} />);
    fireEvent.click(screen.getByRole('button'));
    expect(onClick).toHaveBeenCalledTimes(1);
  });

  it('calls onBlur handler', () => {
    const onBlur = vi.fn();
    render(<Burger onBlur={onBlur} onClick={() => {}} />);
    fireEvent.blur(screen.getByRole('button'));
    expect(onBlur).toHaveBeenCalledTimes(1);
  });
});
