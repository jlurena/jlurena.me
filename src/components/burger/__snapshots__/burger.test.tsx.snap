// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<Burger /> > renders correctly (snapshot) 1`] = `
<DocumentFragment>
  <div
    class="_burgerContainer_a465a2"
  >
    <button
      aria-label="Menu"
      class="_burger_a465a2"
      type="button"
    >
      <div />
      <div />
      <div />
      <div />
    </button>
  </div>
</DocumentFragment>
`;
